pandas/tseries/offsets.py

```diff
@@ -507,8 +513,18 @@ def freqstr(self):
         else:
             fstr = code
 
+        try:
+            if self._offset:
+                fstr += self._offset_str()
+        except AttributeError:
+            # TODO: standardize `_offset` vs `offset` naming convention
+            pass
+
         return fstr
 
+    def _offset_str(self):
```

Comment:
try not to add new things which just clutter up

Reply:
`_offset_str` is currently a method of `BusinessDay`, which has its own implementation of `freqstr`.  Adding this dummy method lets us get rid of the duplicate freqstr method.  Same idea with `__repr__/_repr_attrs`.  Strictly less clutter.